File: Count the Substrings[ 29-March-2023]/Approach 2[ O(n^2) ]/Solution.java
// 	>> JAVA CODE <<  

int countSubstring(String S) 
    { 
        int ans=0;
        for(int i=0;i<S.length();i++)
        {
            int c=0;
            for(int j=i;j<S.length();j++)
            {
                if(Character.isLowerCase(S.charAt(j)))
                    c++;
                else
                    c--;
                    
                if(c==0)
                    ans++;
            }
        }
        return ans;
    }

File: Count the Substrings[ 29-March-2023]/Approach 2[ O(n^2) ]/Solution.cpp
// 		>> C++ CODE <<

int countSubstring(string S)
    {
        int ans=0;
        for(int i=0;i<S.size();i++)
        {
            int c=0;
            for(int j=i;j<S.size();j++)
            {
                if(S[j]>='a'&&S[j]<='z')
                    c++;
                else
                    c--;
                if(c==0)
                    ans++;
            }
        }
        return ans;
    }